Source/SwiftLintFramework/Rules/FunctionParameterCountRule.swift

```diff
@@ -40,6 +45,10 @@ public struct FunctionParameterCountRule: ASTRule, ConfigurationProviderRule {
         let length = Int(dictionary["key.namelength"] as? Int64 ?? 0)
         let substructure = dictionary["key.substructure"] as? [SourceKitRepresentable] ?? []
 
+        if file.contents.substring(nameOffset, length: length).containsString("init(") {
```

Comment:
- `nameOffset` is counted by byte based number.
So, this make failing test with triggering example such as:
```swift
            "struct Foo {\n /* 👨‍👩‍👧‍👦👨‍👩‍👧‍👦👨‍👩‍👧‍👦 */" +
                "init(a: Int, b: Int, c: Int, d: Int, e: Int, f: Int) {}\n" +
                "func bar(a: b, c: Int, d: Int, e: Int, f: Int, g: Int) {}}"
```
Please use SourceKittenFrameworks's byte base API such as `NSString.substringWithByteRange(start:length:)` if getting substring from `file.contents`.
- This also fails with `init?(`.

What I meant on my last comment were:
- "key.name" already contains string what we need here.  
   Most of offset conversion between byte and utf16 are relatively high cost operations. We want to avoid using that if we can.
- And that has prefix `init(` whether initializer is failable or not